types/block.go

```diff
@@ -1400,10 +1400,32 @@ func (data *Data) ComputeShares() (NamespacedShares, int) {
 		tailShares...), curLen
 }
 
-func getNextSquareNum(length int) int {
-	width := int(math.Ceil(math.Sqrt(float64(length))))
-	// TODO(ismail): make width a power of two instead
-	return width * width
+// paddedLen calculates the number of shares needed to make a power of 2 square
+// given the current number of shares
+func paddedLen(length int) int {
+	width := uint32(math.Ceil(math.Sqrt(float64(length))))
+	width = nextPowerOf2(width)
+	return int(width * width)
+}
+
+// nextPowerOf2 returns the next lowest power of 2 unless the input is a power
+// of two, in which case it returns the input
+func nextPowerOf2(v uint32) uint32 {
+	if v == 1 {
+		return 1
+	}
+
+	// find the next highest power using bit mashing
+	v--
```

Comment:
This looks like it might not behave properly if `v == 0`.

Reply:
https://github.com/lazyledger/lazyledger-core/blob/e5a7683c44bbf92d759b127226c0652c7b9018e5/types/block_test.go#L1420

Currently input of 0 results in 0 out, should we be expecting an output of 2?

Reply:
Ah you're right, it does output `0`. That seems fine , since you later check

https://github.com/lazyledger/lazyledger-core/blob/40510a70e92d075ab77b2a13b87d1d53743f684b/types/block.go#L1388

Reply:
added the check and renamed to `nextHighestPowerOfTwo` to be clearer https://github.com/lazyledger/lazyledger-core/pull/331/commits/4aa3372fab1efb976da22e362067b77eeacab138